dotnet/src/IntegrationTests/Connectors/Memory/AzureCosmosDBMongoDB/AzureCosmosDBMongoDBVectorStoreRecordCollectionTests.cs

```diff
@@ -324,9 +325,41 @@ public async Task UpsertWithBsonVectorStoreWithNameModelWorksCorrectlyAsync()
         Assert.Equal("Test Name", getResult.HotelName);
     }
 
+    [Fact(Skip = SkipReason)]
+    public async Task VectorizedSearchWorksCorrectlyAsync()
+    {
+        // Arrange
+        var hotel1 = this.CreateTestHotel(hotelId: "key1", embedding: new[] { 30f, 31f, 32f, 33f });
+        var hotel2 = this.CreateTestHotel(hotelId: "key2", embedding: new[] { 20f, 20f, 20f, 20f });
+        var hotel3 = this.CreateTestHotel(hotelId: "key3", embedding: new[] { 10f, 10f, 10f, 10f });
+        var hotel4 = this.CreateTestHotel(hotelId: "key4", embedding: new[] { 0f, 0f, 0f, 0f });
+
+        var sut = new AzureCosmosDBMongoDBVectorStoreRecordCollection<AzureCosmosDBMongoDBHotel>(fixture.MongoDatabase, "TestVectorizedSearch");
+
+        await sut.CreateCollectionIfNotExistsAsync();
+
+        await sut.UpsertBatchAsync([hotel1, hotel2, hotel3, hotel4]).ToListAsync();
+
+        // Act
+        var searchResults = await sut.VectorizedSearchAsync(new ReadOnlyMemory<float>([30f, 31f, 32f, 33f])).ToListAsync();
+
+        // Assert
+        var ids = searchResults.Select(l => l.Record.HotelId);
+
+        Assert.Contains("key1", ids);
```

Comment:
Might also be good to check that the keys are in the right order, e.g. 30 should be first since it's closest to the request vector.